### ISSUE TITLE
Updateing a big_map inside an asset

### ISSUE THREAD

**Issue by evronm** [completium/archetype-lang#328]
Sorry for asking all these questions, but I promise I'll get the hang of it soon.  Any, here's what I have now:
```
asset tokens {
  symbol: string;
  name: string;
  ledger: iterable_big_map<address, nat >;
} 
entry xfer(sym: string, %from : address, %to : address, amount : nat) {
  require {
    r: (tokens[sym].ledger[%from] ? the : 0) > amount otherwise "INSUFFICIENT_TOKENS"
  }
  effect {
    var x ?= tokens[sym].ledger[%from];
    tokens[sym].ledger.update(%from, x - amount);
  }
}
```

And I get an error on the last line:  No such method "update".  The strange thing is that the previous line works fine.  What do I need to do to access the update method here.

Thanks yet again.

**Comment by 94810** [comment]
If I am not mistaken you need to wrap your value in an `option` type, but you also need to convert your subtraction result from int to nat. `int_to_nat` already return an `option<nat>` so something like this `tokens[sym].ledger.update(%from, int_to_nat(x - amount));`.

**Comment by evronm** [comment]
No, they are both nat.  Also the error is that the update method is unavailable.  I tried anyway, and as I thought, no change in the error message.

**Comment by rognierbenoit** [comment]
Hi @evronm, the issue here comes from the `iterable_big_map` that does not provide `update` indeed. if you change it to `big_map` it should work. `iterable_big_map` is still a bit experimental. so the working example:
```

asset tokens {
   id      : string;
   ledger  : big_map<address, nat> = []
}
entry add(a : address, v : string, n : nat) {
  tokens.add_update(v, { });
  tokens[v].ledger.put(a, n)
}

view get_balance(a : address, v : string) : nat {
  return (tokens[v] ? (the.ledger[a] ? the : 0) : 0)
}
```

**Comment by evronm** [comment]
I thought that might be the case.  Thing is, I need an iterable map here because I need to be able to get a list of accounts.    I could just keep track in a separate set, but then I'm back to the reason I ditched Solidity.  

So I tried:

```
var x ?= tokens[sym].ledger[%from];
tokens[sym].ledger[%from] -=x;
```
and getting a "invalid type for tuple" error.  Then I tried:

```
    var x ?= tokens[sym].ledger[%from];
    tokens[sym].ledger.remove(%from);
    tokens[sym].ledger.put(%from, x- amount);
```
And got a 'No such method: "put"' error.  But I noticed this only happened when there was an operation in the second argument, so I am left with this monstrosity, which actually works:
```
var x ?= tokens[sym].ledger[%from];
x -= amount;
tokens[sym].ledger.remove(%from);
tokens[sym].ledger.put(%from, x);
x += 2 * amount;
tokens[sym].ledger.remove(%to);
tokens[sym].ledger.put(%to, x);
```

It's still better than keeping a separate array, and I can hide it all in a function (I think), but is there a less hideous way to express this in Archtype?

Thanks yet again.
